h2o-core/src/main/java/water/jdbc/SQLManager.java

```diff
@@ -269,26 +306,26 @@ public ConnectionPoolProvider() {} // Externalizable classes need no-args constr
 
     }
 
-    /**
-     * @return Number of connections to an SQL database to be opened on a single node.
-     */
-    static int getMaxConnectionsPerNode(final int cloudSize, final short nThreads, final int nChunks) {
-      int conPerNode;
+    static int getMaxConnectionsTotal() {
+      int maxConnections = MAX_CONNECTIONS;
       final String userDefinedMaxConnections = System.getProperty(MAX_USR_CONNECTIONS_KEY);
       try {
-        Integer connectionAmount = Integer.valueOf(userDefinedMaxConnections);
-        if (connectionAmount > 0 && connectionAmount < MAX_CONNECTIONS) {
-          conPerNode = calculateLocalConnectionCount(connectionAmount, cloudSize, nThreads, nChunks);
-        } else {
-          conPerNode = calculateLocalConnectionCount(MAX_CONNECTIONS, cloudSize, nThreads, nChunks);
+        Integer userMaxConnections = Integer.valueOf(userDefinedMaxConnections);
+        if (userMaxConnections > 0 && userMaxConnections < MAX_CONNECTIONS) {
+          maxConnections = userMaxConnections;
         }
       } catch (NumberFormatException e) {
         Log.info("Unable to parse maximal number of connections: " + userDefinedMaxConnections
-            + ". Falling back to default settings.");
-        conPerNode = calculateLocalConnectionCount(MAX_CONNECTIONS, cloudSize, nThreads, nChunks);
+                + ". Falling back to default settings.");
       }
+      return maxConnections;
+    }
 
-      return conPerNode;
+    /**
+     * @return Number of connections to an SQL database to be opened on a single node.
+     */
+    static int getMaxConnectionsPerNode(final int cloudSize, final short nThreads, final int nChunks) {
+     return calculateLocalConnectionCount(getMaxConnectionsTotal(), cloudSize, nThreads, nChunks);
```

Comment:
This seems to be just a delegate method to my old method. Maybe we can call `getMaxConnectionsTotal()! directly in `calculateLocalConnectionCount! - because the task here is to obtain local connections count per node anyway. The call chains get a little bit confusing. Let's keep just one method - in a way it makes sense to call `getMaxConnectionsTotal`in calculateLocalConnectionCount, because in order to calculate the number of local connections, the maximal number of connections must be known.

Reply:
I've just split the original getMaxConnectionsPerNode, because it was doing 2 different things:
1. reading the *global* property: max_connections param/defaults: this is extracted to getMaxConnectionsTotal().
2. computing the *local* connections needed for a single node to fetch data.
Therefore, I think separation of concerns applies here.
Besides, #1 is also needed by getOptimalConnectionPerNode below.

Reply:
Oh I see, if you want me to add your existing calculateLocalConnectionCount logic, into this method, then I'll rather do it in the refactoring branch: people don't like when refactorings are mixed with logic changes :)

Reply:
closed